view/js/main.js

```diff
@@ -33,6 +33,50 @@ function _resizeIframe(obj, desth) {
 	setTimeout(_resizeIframe, 100, obj, ch);
 }
 
+function getCookie(name) {
+	var value = "; " + document.cookie;
+	var parts = value.split("; " + name + "=");
+
+	if (parts.length == 2) {
+		return parts.pop().split(";").shift();
+	}
+}
+
+function initWidget(inflated, deflated) {
+	var elInf = document.getElementById(inflated);
+	var elDef = document.getElementById(deflated);
+
+	if (!elInf || !elDef) {
+		return;
+	}
+	if (getCookie(window.location.pathname + ":" + inflated) != "none") {
+		elInf.style.display = "block";
+		elDef.style.display = "none";
+	} else {
+		elInf.style.display = "none";
+		elDef.style.display = "block";
+	}
+}
+
+function openCloseWidget(inflated, deflated) {
```

Comment:
I'd prefer you used an event listener for this instead of explicitly referencing the Javascript function in the `onclick` attribute of the fake links.

Vanilla Javascript:
```javascript
widgetFakelink.addEventListener('click', function () {
   ...
});
```
In JQuery:
```javascript
$('body').on('click', '.widget .fakelink', function () {
   ...
});
```

Reply:
We use this "onclick" in many places, just with a different function. Why not here?

Reply:
And I would argue we shouldn't either in the other places. And while I can't just replace all the existing attribute occurrences with event listeners, I can ask for changes on new additions.

The problem is that it tightly couples the HTML and the Javascript. I prefer when HTML files contain only HTML. Since base templates can be used by multiple themes, we might want different behaviors by theme. With the `onclick` attribute, we can't really do that.

Reply:
The whole HTML template can only work with the JS functionality. So splitting doesn't make any sense. Also the element is marked as "fakelink" which implies the need for an "onclick" event there.

I also see no benefit in splitting the JS to some other file where we had to specify each single element id (we cannot do this for all "widget" elements) as well.

Reply:
You don't have to specify every single element, you can use a specific class for it, like `widget-collapsible` for example.

Ideally the HTML should stand as is and you only add enhancement with Javascript when it's completely loaded. It's an accessibility feature where you shouldn't depend on Javascript at all and the onclick attribute go against this.

Reply:
Well, stuff like that is far beyond my JS horizon. If someone else was developing such a concept for some other place in Friendica then I might be able to adapt this to this case here, but that's all.

Reply:
I can do it, please merge https://github.com/friendica/friendica/pull/8085 and I'll take over this to do it the right way.

Reply:
My suggestion would be, merging this PR and later enhancing it. This has got the advantage that the feature isn't removed until it is reimplemented.

Reply:
I guess I have to pick my battles.